File: ios/Classes/entity/FriendEntity.swift
import ImSDK;

//  Created by 蒋具宏 on 2020/2/13.
//  朋友实体
public class FriendEntity : NSObject{
    
    /**
     * ID
     */
    var identifier : String?;
    
    /**
     *  添加来源
     */
    var addSource : String?;
    
    /**
     *  添加时间
     */
    var addTime : UInt64?;
    
    /**
     *  添加说明
     */
    var addWording : String?;
    /**
     *  备注
     */
    var remark : String?;
    /**
     *  用户信息
     */
    var timUserProfile : UserInfoEntity?;
    
    override init() {
    }
    
    init(friend : TIMFriend) {
        super.init();
        self.identifier = friend.identifier;
        self.addSource = friend.addSource;
        self.addTime = friend.addTime;
        self.addWording = friend.addWording;
        self.remark = friend.remark
        self.timUserProfile = UserInfoEntity(userProfile: friend.profile);
    }
}

File: lib/entity/friend_entity.dart
import 'package:tencent_im_plugin/entity/user_info_entity.dart';

/// 好友实体
class FriendEntity {
  /// ID
  String identifier;

  /// 添加来源
  String addSource;

  /// 添加时间
  int addTime;

  /// 申请描述
  String addWording;

  /// 备注
  String remark;

  /// 用户信息
  UserInfoEntity userInfoEntity;

  FriendEntity({
    this.identifier,
    this.addSource,
    this.addTime,
    this.addWording,
    this.remark,
    this.userInfoEntity,
  });

  FriendEntity.fromJson(Map<String, dynamic> json) {
    identifier = json['identifier'];
    addSource = json['addSource'];
    addTime = json['addTime'];
    addWording = json['addWording'];
    remark = json['remark'];
    userInfoEntity = json['timUserProfile'] == null
        ? null
        : UserInfoEntity.fromJson(json['timUserProfile']);
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['identifier'] = this.identifier;
    data['addSource'] = this.addSource;
    data['addTime'] = this.addTime;
    data['addWording'] = this.addWording;
    data['remark'] = this.remark;
    if (this.userInfoEntity != null) {
      data['timUserProfile'] = this.userInfoEntity.toJson();
    }
    return data;
  }

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is FriendEntity &&
          runtimeType == other.runtimeType &&
          identifier == other.identifier;

  @override
  int get hashCode => identifier.hashCode;
}
